control.lua

```diff
@@ -71,7 +86,7 @@ end
 
 local function on_pre_entity_removed(event)
   local entity = event.entity
-  if entity.name == PC_ENTITY_NAME then
+  if entity and entity.valid and entity.name == PC_ENTITY_NAME then
```

Comment:
does entity.valid guard entity.name?  that is can you access the name if valid is false?

this more stringent check might leave invalid entities registered and prevent them from being garbage collected

Reply:
the data map uses the `unit_number` so as long as the that is available on entities which are not valid they can still be unregistered/cleared from the table.

I agree that not crashing is the number 1 priority.  I think this code is fine to go in.